### ISSUE TITLE
Lint/AmbiguousOperator registers offense for macro methods

### ISSUE THREAD

**Issue by Drenmi** [rubocop-hq/rubocop#4294]
The `Lint/AmbiguousOperator` cop will register an offense when a splatted argument is passed to a macro method. This usage should probably not be considered ambiguous, because it doesn't make sense to have a multiplication expression in the `class`- or `module` body context.

--------

## Expected behavior

RuboCop does not register an offense.

## Actual behavior

RuboCop registers an offense.

## Steps to reproduce the problem

Run RuboCop on:

```ruby
class Foo
  attr_reader *BAR
end
```

## RuboCop version

```
$ rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by mikegee** [comment]
🤔  Would it become ambiguous if it were in an assignment?

```ruby
class Foo
  BAR = dsl_method *args
end
```

**Comment by Drenmi** [comment]
@mikegee: Yes. That would be ambiguous since the evaluated expression is used. 🙂

**Comment by Drenmi** [comment]
Looks like this cop is just a wrapper around Parser diagnostics. I'm unsure if there's a straightforward way of making this exception without re-implementing the cop. 🤔

**Comment by bbatsov** [comment]
I guess another option would be to change this in `parser`.
